flutter/cpp/datasets/coco_gen.cc

```diff
@@ -0,0 +1,108 @@
+/* Copyright 2024 The MLPerf Authors. All Rights Reserved.
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+==============================================================================*/
+
+#include "coco_gen.h"
+
+#include <iomanip>
+
+namespace mlperf {
+namespace mobile {
+namespace {}  // namespace
+
+CocoGen::CocoGen(Backend* backend, const std::string& input_tfrecord,
```

Comment:
Hi @anhappdev , 

do we have the Accuracy dataset and the input_clip_model shared anywhere? can you please share the link?

Reply:
The files are stored in Google Drive linked in the first comment of this PR.

Reply:
@anhappdev, what should be the input format for the clip_model.tflite? is it required to be NHWC or NCHW?

Reply:
The `clip_model_512x512.tflite` is the one to be used by the app, and the input should be 512x512 image in NHWC layout.
You can see the conversion script at `flutter/cpp/datasets/coco_gen_utils/CLIP_Model_to_TFLite.ipynb`

Reply:
do we have the tfrecord file decided for the validation set?

Reply:
No, not yet. But you can easy generate it with the script at `flutter/cpp/datasets/coco_gen_utils/generate_tfrecords.py`

Related comment:
https://github.com/mlcommons/mobile_model_closed/issues/9#issuecomment-2319660564

Reply:
I uploaded a `coco_gen_test.tfrecord` with 150 prompts to the Google Drive folder.